composer/tools/copy_environment.py

```diff
@@ -0,0 +1,761 @@
+from google.oauth2 import service_account
+from googleapiclient import discovery, errors
+from kubernetes import client, config
+from mysql import connector
+import six
+from six.moves import configparser
+
+DEFAULT_SCOPES = ["https://www.googleapis.com/auth/cloud-platform"]
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        description="Clone a composer environment."
+    )
+    parser.add_argument(
+        "project",
+        help="Google Cloud Project containing existing Composer Environment.",
+    )
+    parser.add_argument(
+        "location",
+        help="Google Cloud region containing existing Composer Environment. "
+        "For example `us-central1`.",
+    )
+    parser.add_argument(
+        "existing_env_name",
+        help="The name of the existing Composer Environment.",
+    )
+    parser.add_argument(
+        "new_env_name",
+        help="The name to use for the new Composer Environment.",
+    )
+    parser.add_argument(
+        "--running_as_service_account",
+        action="store_true",
+        help="Set this flag if the script is running on a VM with same "
+        "service account as used in the Composer Environment. This avoids "
+        "creating extra credentials.",
+    )
+    parser.add_argument(
+        "--override_machine_type",
+        default=None,
+        help="Optional. Overrides machine type used for Cloud Composer "
+        "Environment. Must be a fully specified machine type URI.",
+    )
+    parser.add_argument(
+        "--override_disk_size_gb",
+        type=int,
+        default=0,
+        help="Optional. Overrides the disk size in GB used for Cloud Composer "
+        "Environment.",
+    )
+    parser.add_argument(
+        "--override_network",
+        default=None,
+        help="Optional. Overrides the network used for Cloud Composer "
+        "Environment.",
+    )
+    parser.add_argument(
+        "--override_subnetwork",
+        default=None,
+        help="Optional. Overrides the subnetwork used for Cloud Composer "
+        "Environment.",
+    )
+    parser.add_argument(
+        "--override_service_account",
+        default=None,
+        help="Optional. Overrides the service account used for Cloud Composer "
+             "Environment.",
+    )
+    parser.add_argument(
+        "--override_beta_api",
+        default=False,
+        help="Optional. Overrides the service API used for Cloud Composer "
+             "Environment. If true then Beta API will be used",
+    )
+    return parser.parse_args()
+
+
+def get_composer_env(composer_client, project_id, location, name):
+    request = (
+        composer_client.projects()
+        .locations()
+        .environments()
+        .get(
+            name="projects/{}/locations/{}/environments/{}".format(
+                project_id, location, name
+            )
+        )
+    )
+    return request.execute()
+
+
+def wait_composer_operation(
+    composer_client, operation_name, exit_on_error=True
```

Comment:
Maybe add a docstring for the method description, e.g.:

"""Waits for the operation to composer operation to complete and reports failures."""